data_board/common/lithium.h

```diff
@@ -24,43 +37,437 @@ typedef enum lithium_result {
+ *
+ * @param radio The radio to communicate with
+ * @param data The contents of the beacon
+ * @param length The length of the beacon contents
+ *
+ * @return The result of the operation
+ */
+lithium_result_t lithium_send_set_beacon(lithium_t * radio, uint8_t * data, uint16_t length);
+
+/**
+ * Send a beacon configuration change packet to a Lithium radio
+ *
+ * @param radio The radio to communicate with
+ * @param config The configuration to send to the beacon module
+ *
+ * @return The result of the operation
+ */
+lithium_result_t lithium_send_set_beacon_config(lithium_t * radio, lithium_beacon_config_t * config);
+
+/**
+ * Send a firmware version request packet to a Lithium radio
+ *
+ * @param radio The radio to communicate with
+ *
+ * @return The result of the operation
+ */
+lithium_result_t lithium_send_read_fw_version(lithium_t * radio);
+
+/**
+ * Send an over-the-air key change packet to a Lithium radio
+ *
+ * @param radio The radio to communicate with
+ * @param key The 4 byte key to change to
+ *
+ * @return The result of the operation
+ */
+lithium_result_t lithium_send_write_dio_key(lithium_t * radio, uint8_t * key);
+
+/**
+ * Send a firmware update initiation packet to a Lithium radio
+ *
+ * @param radio The radio to communicate with
+ * @param hash The 16 byte MD5 hash of the firmware update
+ *
+ * @return The result of the operation
+ */
+lithium_result_t lithium_send_begin_fw_update(lithium_t * radio, uint8_t * hash);
+
+/**
+ * Send a firmware update chunk packet to a Lithium radio
+ *
+ * @param radio The radio to communicate with
+ * @param data The chunk of the firmware update
+ * @param length The length of the chunk
+ *
+ * @return The result of the operation
+ */
+lithium_result_t lithium_send_stream_fw_update(lithium_t * radio, uint8_t * data, uint16_t length);
 
 /**
- * Send a noop command to an open lithium radio
+ * Send a power amplifier level change packet to a Lithium radio
  *
  * @param radio The radio to communicate with
+ * @param speed The speed to change the power amplifier level to
+ *
+ * @return The result of the operation
+ */
+lithium_result_t lithium_send_set_pa_level(lithium_t * radio, uint8_t speed);
+
+/**
+ * Attempt to receive and parse a Lithium packet over the UART channel
+ *
+ * @param radio The radio to communicate with
+ * @param packet The output packet parsed from the channel
+ *
+ * @return The result of the operation
+ */
+lithium_result_t lithium_receive_packet(lithium_t * radio, lithium_packet_t * packet);
+
+/**
+ * Parse the header of an encoded Lithium packet
+ *
+ * @param raw_packet The encoded packet header to parse
+ * @param packet The output packet with header fields populated
+ * @param remaining_bytes The output number of bytes to read to parse a body
+ *
+ * @return The result of the operation
+ */
+lithium_result_t lithium_parse_header(uint8_t * raw_packet, lithium_packet_t * packet, uint16_t * remaining_bytes);
```

Comment:
Should we also pass a buffer length here? I know the packet header should have a length but if that's corrupt/we missed part of the packet it would be nice to have some cross checking.